patches/chrome-installer-mini_installer-mini_installer.cc.patch

```diff
@@ -1,192 +1,12 @@
-+  PathString filename;
-+  if (!filename.assign(
-+          GetNameFromPathExt(installer_filename, lstrlen(installer_filename))))
-+    return false;
-+
-+  // Strip extension from filename.
-+  const wchar_t* scan = filename.get() + filename.length() - 1;
-+  while (scan != filename.get() && *scan != L'.')
-+    --scan;
-+
-+  if (*scan == L'.')
-+    filename.truncate_at(scan - filename.get());
-+
-+  // Strip any de-duplicating suffix from filename, e.g. "(1)".
-+  scan = filename.get() + filename.length() - 1;
-+  if (*scan == L')') {
-+    --scan;
-+    while (scan != filename.get() && *scan >= '0' && *scan <= '9')
-+      --scan;
-+    if (*scan == L'(')
-+      filename.truncate_at(scan - filename.get());
-+  }
-+
-+  // Strip trailing spaces from filename.
-+  scan = filename.get() + filename.length() - 1;
-+  while (scan != filename.get() && *scan == L' ')
-+    --scan;
-+
-+  if (scan != filename.get() && (scan != filename.get() + filename.length()))
-+    filename.truncate_at(scan - filename.get() + 1);
-+
-+  // First check for 6-character standard referral code XXXDDD, where
-+  // X is an alphabetic character and D is a numeric character. If not
-+  // found, check for an alphabetic referral code of any length in the
-+  // form XXX-XXX.
-+  if (!ParseStandardReferralCode(filename.get(), referral_code) &&
-+      !ParseExtendedReferralCode(filename.get(), referral_code)) {
-+    return false;
-+  }
-+
-+  return true;
-+}
-+
-+bool GetStubInstallerFilename(PathString& installer_filename) {
-+  wchar_t value[MAX_PATH] = {0,};
-+  const bool result =
-+      RegKey::ReadSZValue(HKEY_CURRENT_USER, L"Software\\BraveSoftware\\Promo",
-+                          L"StubInstallerPath", value, _countof(value)) ;
-+
-+  if (!result)
-+    return false;
-+
-+  if (!installer_filename.assign(value) || installer_filename.length() == 0)
-+    return false;
-+
-+  return true;
-+}
-+#endif  // BRAVE_CHROMIUM_BUILD
-+
- // Gets the setup.exe path from Registry by looking at the value of Uninstall
- // string.  |size| is measured in wchar_t units.
- ProcessExitResult GetSetupExePathForAppGuid(bool system_level,
-@@ -528,6 +660,21 @@ ProcessExitResult RunSetup(const Configuration& configuration,
+@@ -528,6 +528,7 @@ ProcessExitResult RunSetup(const Configuration& configuration,
    // on to setup.exe
    AppendCommandLineFlags(configuration.command_line(), &cmd_line);
  
-+#if defined(BRAVE_CHROMIUM_BUILD)
-+  PathString installer_filename;
-+  if (GetStubInstallerFilename(installer_filename)) {
-+    ReferralCodeString referral_code;
-+    if (ParseReferralCode(installer_filename.get(), referral_code)) {
-+      if (!cmd_line.append(L" --") ||
-+          !cmd_line.append(L"brave-referral-code") ||
-+          !cmd_line.append(L"=\"") || !cmd_line.append(referral_code.get()) ||
-+          !cmd_line.append(L"\"")) {
-+        return ProcessExitResult(COMMAND_STRING_OVERFLOW);
-+      }
-+    }
-+  }
-+#endif
-+
++  BRAVE_RUN_SETUP
```

Comment:
@emerick why are doing this after `AppendCommandLineFlags`?

Reply:
It's been a while, but I think I had some feedback on the original issue that our option should be last so that if we exceedeed the command-line length only our option would be affected.